pkg/controller/stacks/install/stackinstall.go

```diff
@@ -237,6 +251,38 @@ func (h *stackInstallHandler) update(ctx context.Context) (reconcile.Result, err
 	return reconcile.Result{}, nil
 }
 
+// delete performs clean up (finalizer) actions when a StackInstall is being deleted.
+// This function ensures that all the resources (e.g., CRDs) that this StackInstall owns
+// are also cleaned up.
+func (h *stackInstallHandler) delete(ctx context.Context) (reconcile.Result, error) {
+	gvk := h.ext.GroupVersionKind()
+	labels := map[string]string{
+		labelParentGroup:     gvk.Group,
+		labelParentVersion:   gvk.Version,
+		labelParentKind:      gvk.Kind,
+		labelParentNamespace: h.ext.GetNamespace(),
+		labelParentName:      h.ext.GetName(),
+		labelParentUID:       string(h.ext.GetUID()),
+	}
+	list := &apiextensionsv1beta1.CustomResourceDefinitionList{}
+	if err := h.kube.List(ctx, list, client.MatchingLabels(labels)); err != nil {
+		return fail(ctx, h.kube, h.ext, err)
+	}
+
+	for i := range list.Items {
+		if err := h.kube.Delete(ctx, &list.Items[i]); runtimeresource.IgnoreNotFound(err) != nil {
+			return fail(ctx, h.kube, h.ext, err)
+		}
+	}
+
+	meta.RemoveFinalizer(h.ext, installFinalizer)
```

Comment:
Do we have to, or should we filter, on `installFinalizer` before performing all of these delete specific actions? 

I'm not familiar enough with the delete and finalizer flow.  While I don't doubt that this code would work, do we end up doing double lookup/delete work if this `delete` method is called multiple times, perhaps because multiple finalizers are in place?

Reply:
good question.  filtering on finalizer isn't strictly necessary because the main purpose of the finalizer is to block deletion of the object from the API server until this controller has finished its deletion work and clear the finalizer.  If the object has a deletion timestamp, it's fair game for running the deletion handling logic.

I think we used to filter by finalizer earlier in this project's history, but the contemporary patterns do not, and it's not strictly necessary.

All deletion operations should be idempotent and can be called multiple times.